File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {numbers[]} numbers
     * @param {numbers} target
     * @return {numbers[]}
     */
    twoSum(numbers, target) {
        let l = 0;
        let r = numbers.length - 1;
        while (numbers[l] + numbers[r] != target) {
            if (numbers[l] + numbers[r] > target) {
                r--;
            } else {
                l++;
            }
        }
        console.log([numbers[l],numbers[r]])
        return [l+1,r+1]
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        merge = []
        for sublist in matrix:
            merge += sublist 
        l, r = 0, len(merge) - 1
        while l <= r:
            m = l + (r - l) // 2
            if merge[m] == target:
                return True
            elif merge[m] < target:
                l = m + 1
            elif merge[m] > target:
                r = m - 1
        return False